client-side/src/components/date/GenericDate.jsx

```diff
@@ -0,0 +1,83 @@
+
+import React, { useState } from 'react';
+import axios from 'axios';
+import Select from '../../stories/Select/Select.jsx';
+import GenericButton from '../../stories/Button/GenericButton.jsx';
+import PropTypes from 'prop-types';
+import formatDate from './formatDate.js';
+
+const DateFormatter = ({ currentUser, date }) => {
+  const { emailFrequency, sendNotificationTime, soundVoice, _id } = currentUser.preferences;
+  const { formatedDate } = currentUser;
+  const [dateFormat, setDateFormat] = useState(formatedDate || 'yyyy-MM-dd');
+
+  const url = process.env.REACT_APP_BASE_URL;
+
+  const dateFormats = [
+    { value: 'yyyy-MM-dd', label: 'yyyy-MM-dd' },
```

Comment:
the formats should be:
yyyy-mm-dd
mm-dd-yyyy
dd-mm-yyyy

and they should be declared in constants file